compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeLayer.desktop.kt

```diff
@@ -74,21 +72,32 @@ import java.awt.event.WindowEvent
 import java.awt.event.WindowFocusListener
 import java.awt.im.InputMethodRequests
 import javax.accessibility.Accessible
-import javax.accessibility.AccessibleContext
 import javax.swing.SwingUtilities
 import kotlin.coroutines.AbstractCoroutineContextElement
 import kotlin.coroutines.CoroutineContext
-import java.awt.Cursor
-import java.awt.event.FocusListener
+import kotlinx.coroutines.CoroutineExceptionHandler
+import org.jetbrains.skia.Canvas
+import org.jetbrains.skiko.MainUIDispatcher
+import org.jetbrains.skiko.SkiaLayer
 import org.jetbrains.skiko.SkiaLayerAnalytics
-import org.jetbrains.skiko.hostOs
 import org.jetbrains.skiko.SkikoInput
+import org.jetbrains.skiko.SkikoView
+import org.jetbrains.skiko.hostOs
 
 internal class ComposeLayer(
     private val skiaLayerAnalytics: SkiaLayerAnalytics
 ) {
     private var isDisposed = false
 
+    internal val sceneAccessible = ComposeSceneAccessible(
+        rootsProvider = {
+            scene.roots.filterIsInstance<SkiaBasedOwner>()
```

Comment:
This and the code in `ComposeSceneAccessible` could be improved for performance and clarity, as we discussed.
The suggestion is:
1. Improve `ComposeLayer.list`:
  1.1 Rename it (and `listCopy`) to `owners`
  1.2 Make it a `MutableList` and document that the order matters for the accessibility search.
  1.3 `ComposeScene.roots` can return `owners.toSet()`. Check with @igordmn to be sure it's ok.
2. Pass `owners.asReversed()` to `ComposeSceneAccessible.rootsProvider` and get rid of the extra list allocations in `ComposeSceneAccessible`.